test/lib/tasks/minify.integration.js

```diff
@@ -341,3 +345,81 @@ return;`;
 			`/resources/my/namespace/test.js (line 3, col 0, pos 48)`
 	}, `Threw with expected error message`);
 });
+
+test.serial("integration: minify with taskUtil and resources tagged with OmitFromBuildResult", async (t) => {
+	const {reader, writer, workspace} = createWorkspace();
+
+	const testFilePath1 = "/resources/my/namespace/test1.js";
+	const testFilePath2 = "/resources/my/namespace/test2.js";
+	const testFileContent1 = "function test(param1) { var variableA = param1; console.log(variableA); } test();";
+	const testFileContent2 = "function test(param2) { var variableB = param2; console.log(variableB); } test();";
+
+	const testResource1 = resourceFactory.createResource({
+		path: testFilePath1,
+		string: testFileContent1
+	});
+	const testResource2 = resourceFactory.createResource({
+		path: testFilePath2,
+		string: testFileContent2
+	});
+
+	await reader.write(testResource1);
+	await reader.write(testResource2);
+
+	const taskUtil = {
+		STANDARD_TAGS: {
+			HasDebugVariant: "1️⃣",
+			IsDebugVariant: "2️⃣",
+			OmitFromBuildResult: "3️⃣"
+		},
+		setTag: t.context.sinon.stub(),
+		getTag: t.context.sinon.stub().callsFake((...args) => {
+			if (args[0].getPath().includes(testFilePath1.split(".js").shift()) &&
+				args[1] === taskUtil.STANDARD_TAGS.OmitFromBuildResult) {
+				return true; // OmitFromBuildResult for testFilePath1
+			}
+			return false; // No OmitFromBuildResult for testFilePath2
+		}),
+		registerCleanupTask: t.context.sinon.stub()
+	};
+
+	await minify({
+		workspace,
+		taskUtil,
+		options: {
+			pattern: "/**/*.js",
+		}
+	});
+
+	for (const {filePath, expectToBeOmitted} of [
+		{filePath: testFilePath1, expectToBeOmitted: true},
+		{filePath: testFilePath2, expectToBeOmitted: false}
+	]) {
+		let res = await writer.byPath(filePath);
+		if (!res) {
+			t.fail(`Did not find ${filePath} in target locator although it should be present`);
+		}
+		const fileName = filePath.split("/").pop();
+		const expected = `function test(t){var o=t;console.log(o)}test();\n${SOURCE_MAPPING_URL}=${fileName}.map`;
+		const content = await res.getString();
+		t.deepEqual(content, expected, "Correct file content");
+
+		let isOmitted = taskUtil.getTag(res, taskUtil.STANDARD_TAGS.OmitFromBuildResult);
```

Comment:
I'm afraid calling the stubbed `getTag` method will always return the expected result, no matter whether minify actually called `setTag` correctly. We need to check whether the `setTag` stub got called correctly.

I will create a commit to update this test